sound/soc/sof/trace.c

```diff
@@ -97,10 +110,22 @@ static ssize_t sof_dfsentry_trace_read(struct file *file, char __user *buffer,
 	return count;
 }
 
+static int sof_dfsentry_trace_release(struct inode *inode, struct file *file)
+{
+	struct snd_sof_dfsentry *dfse = inode->i_private;
+	struct snd_sof_dev *sdev = dfse->sdev;
+
+	if (!sdev->dtrace_is_enabled)
+		sdev->host_offset = 0;
+
+	return 0;
+}
+
 static const struct file_operations sof_dfs_trace_fops = {
 	.open = simple_open,
 	.read = sof_dfsentry_trace_read,
 	.llseek = default_llseek,
+	.release = sof_dfsentry_trace_release,
```

Comment:
this looks odd since we don't have a matching function where host_offset is initialized to zero? I would also add this as a separate patch to separate out the changes.

Reply:
@plbossart Here the debugfs versus fwtrace state confusion is apparent. 'host_offset' is controlled by the DMA trace state machine.

If debugfs node is opened while DMA trace is active, we don't really have good solutions how to sync the positions (debufgs client will get data from 0....host_offset).

If debugfs node is opened while DMA trace is not active, we don't want to reset 'host_offset' as the user may want to read whatever traces are in the log (@keyonjie commented on this earlier).

But by resetting 'host_offset' in release when DMA trace is not active, we can at least ensure the same data is passed to client twice.

So long story short, this is a bit iffy, but is there for a reason. I'll add a comment to this, let's see how it reads out.